### DISCUSSION TITLE
Easier CustomShader-Node Boilerplate QOL

### DISCUSSION THREAD

**Discussion by strangerattractor** [cables-gl/cables_docs#735] [Feature Requests] [answered]
Whenever I start writing a shader I always look for these things again and again: 
- how to add a uniform
- how to add a texture sampler

Easy fix proposal for QOL: 
Just use the boilerplate code from Pandurs Texture Shader example: 
https://cables.gl/p/H_zHBm

It has most of the great basic starting stuff:
--------------------------------------- (Sorry, Markdown didn't work for me)

uniform float uResX;
uniform float uResY;

uniform float uTime;

uniform sampler2D mytexture;

IN vec2 texCoord;

{{MODULES_HEAD}}
void main()
{
    vec2 uv = gl_FragCoord.xy/vec2(uResX,uResY);
    
    float animGreen = 0.5 + (0.5 * sin(uTime));
    
    vec4 col=vec4(uv,animGreen,1.0);
    
    col*=texture(mytexture,texCoord);
    
    {{MODULE_COLOR}}
    outColor = col;
}

**Comment by pandrr** [comment, marked as answer]
added some snippets to the editor for those cases, so it can be found easily with the intellisense completion, if you have good ones, let me know

dunno, but this is also not so hard to memorize :>

<img width="314" alt="Screenshot 2024-04-26 at 16 39 59" src="https://github.com/cables-gl/cables_docs/assets/98792/b5e0ecc1-dedd-4638-bdc2-302c905c8645">
<img width="389" alt="Screenshot 2024-04-26 at 16 38 23" src="https://github.com/cables-gl/cables_docs/assets/98792/15069c3f-4fc1-4334-8f48-31f10de6263b">
<img width="523" alt="Screenshot 2024-04-26 at 16 37 25" src="https://github.com/cables-gl/cables_docs/assets/98792/73fe6c7e-88b7-428d-bf8f-0eb987b02115">
